fedimint-server/src/config/api.rs

```diff
@@ -185,21 +186,46 @@ impl ConfigGenApi {
     /// Once configs are generated, starts DKG and await its
     /// completion, calling a second time will return an error
     pub async fn run_dkg(&self) -> ApiResult<()> {
-        // Update our state to running DKG
-        let request = self.get_requested_params()?;
-        let response = self.get_consensus_config_gen_params(&request).await?;
-        let (params, registry) = {
+        let leader = {
             let mut state = self.require_status(ServerStatus::SharingConfigGenParams)?;
+            // Update our state
             state.status = ServerStatus::ReadyForConfigGen;
-            (
-                state.get_config_gen_params(&request, response.consensus)?,
-                state.settings.registry.clone(),
-            )
+            // Create a WSClient for the leader
+            state.local.clone().and_then(|local| {
+                local
+                    .leader_api_url
+                    .map(|url| WsAdminClient::new(url, PeerId::from(0)))
+            })
         };
+
         self.update_leader().await?;
 
+        // Followers wait for leader to signal readiness for DKG
+        if let Some(client) = leader {
+            loop {
+                let status = client.status().await.map_err(|_| {
+                    ApiError::not_found("Unable to connect to the leader".to_string())
+                })?;
```

Comment:
This should probably become more robust in the future, so a short internet disconnection doesn't kill the whole setup process.